apps/gitness/src/pages-v2/pull-request/hooks/usePRChecksDecision.ts

```diff
@@ -0,0 +1,181 @@
+      //   which is error and it is then set for it to be used in the check tab.
+      // *******
+      if (_count.error) {
+        _status = ExecutionState.ERROR
+        setColor('text-destructive')
+        setBackground('text-destructive')
+        setMessage(`${_count.error}/${total} ${pluralize('check', _count.error)} errored out.`)
+      } else if (_count.failure) {
+        _status = ExecutionState.FAILURE
+        setColor('text-destructive')
+        setBackground('text-destructive')
+        setMessage(`${_count.failure}/${total} ${pluralize('check', _count.failure)} failed.`)
+      } else if (_count.killed) {
+        _status = ExecutionState.KILLED
+        setColor('text-destructive')
+        setBackground('text-destructive')
+        setMessage(`${_count.killed}/${total} ${pluralize('check', _count.killed)} killed.`)
+      } else if (_count.running) {
+        _status = ExecutionState.RUNNING
+        setColor('text-warning')
+        setBackground('text-warning')
+        setMessage(`${_count.running}/${total} ${pluralize('check', _count.running)} running.`)
+      } else if (_count.pending) {
+        _status = ExecutionState.PENDING
+        setColor('text-tertiary-background')
+        setBackground('text-tertiary-background')
+        setMessage(`${_count.pending}/${total} ${pluralize('check', _count.pending)} pending.`)
+      } else if (_count.skipped) {
+        _status = ExecutionState.SKIPPED
+        setColor('text-tertiary-background')
+        setBackground('text-tertiary-background')
+        setMessage(`${_count.skipped}/${total} ${pluralize('check', _count.skipped)} skipped.`)
+      } else if (_count.success) {
+        _status = ExecutionState.SUCCESS
+        setColor('text-success')
+        setBackground('text-success')
+        setMessage(`${_count.success}/${total} ${pluralize('check', _count.success)} succeeded.`)
+      }
+
+      setComplete(!_count.pending && !_count.running)
+    } else {
+      setComplete(false)
+    }
+
+    return _status
+  }, [data])
+
+  useEffect(() => {
+    let tornDown = false
+    const pollingFn = () => {
+      if (repoMetadata?.path && pullReqMetadata?.source_sha && !complete && !tornDown) {
+        // TODO: fix racing condition where an ongoing refetch of the old sha overwrites the new one.
+        // TEMPORARY SOLUTION: set debounce to 1 second to reduce likelyhood
+        // refetch({ debounce: 1 }).then(() => { -- theres no debounce on refetches rn
+        refetch().then(() => {
+          if (!tornDown) {
+            interval = window.setTimeout(pollingFn, POLLING_INTERVAL)
+          }
+        })
+      }
+    }
+    let interval = window.setTimeout(pollingFn, POLLING_INTERVAL)
+    return () => {
+      tornDown = true
+      window.clearTimeout(interval)
+    }
+  }, [repoMetadata?.path, pullReqMetadata?.source_sha, complete]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Gotta say it took me about 5 read-throughs of this `useEffect` to fully grasp what it's doing and how it's doing it.

Reply:
uh oh i'll put a todo to refactor this